server_installation/topics/operating-mode/crossdc/proc-setting-up-infinispan.adoc

```diff
@@ -0,0 +1,160 @@
+keytool -importcert -keystore truststore.jks -alias server -file server.crt
+rm server.crt
+----
+
+.Procedure
+
+. Open `infinispan.xml` for editing.
++
+By default, {jdgserver_name} Server uses `server/conf/infinispan.xml` for static configuration such as cluster transport and security mechanisms.
++
+. Configure the {jdgserver_name} cluster transport to perform cross-site replication.
+.. Add the RELAY2 protocol to a JGroups stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<jgroups>
+   <stack name="xsite" extends="udp"> <1>
+      <relay.RELAY2 site="LON" <2>
+                    max_site_masters="1000"/> <3>
+      <remote-sites default-stack="tcp"> <4>
+         <remote-site name="LON"/>
+         <remote-site name="NYC"/>
+      </remote-sites>
+   </stack>
+</jgroups>
+----
+<1> Creates a stack named `xsite` that extends the default UDP cluster transport.
+<2> Adds the RELAY2 protocol and names the cluster you are configuring as `LON`. +
+The site name must be unique to each {jdgserver_name} cluster.
+<3> Sets 1000 as the number of relay nodes for the cluster. You should set a value that is equal to or greater than the maximum number of nodes in your {jdgserver_name} cluster.
+<4> Names all {jdgserver_name} clusters that backup caches with {jdgserver_name} data and uses the default TCP stack for inter-cluster transport.
++
+.. Configure the {jdgserver_name} cluster transport to use the stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<cache-container name="default" statistics="true">
+      <transport cluster="${infinispan.cluster.name:cluster}"
+                 stack="xsite"/> <1>
+</cache-container>
+----
+<1> Uses the `xsite` stack for the cluster.
++
+. Configure the keystore as an SSL identity in the server security realm.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<server-identities>
+  <ssl>
+    <keystore path="server.jks" <1>
+              relative-to="infinispan.server.config.path"
+              keystore-password="password" <2>
+              alias="server" /> <3>
+  </ssl>
+</server-identities>
+----
+<1> Specifies the path of the keystore that contains the SSL identity.
+<2> Specifies the password to access the keystore.
+<3> Names the alias of the certificate in the keystore.
++
+. Configure the authentication mechanism for the Hot Rod endpoint.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<endpoints socket-binding="default" security-realm="default">
+   <hotrod-connector name="hotrod">
+      <authentication security-realm="default">
+         <sasl mechanisms="DIGEST-MD5 GSSAPI EXTERNAL" <1>
+               server-name="infinispan" /> <2>
+      </authentication>
+   </hotrod-connector>
+   <rest-connector name="rest"/>
+</endpoints>
+----
+<1> Configures the SASL authentication mechanism for the Hot Rod endpoint. You should use an appropriate mechanism for your environment. DIGEST-MD5 only is suitable for test and development environments.
```

Comment:
```suggestion
<1> Configures the SASL authentication mechanism for the Hot Rod endpoint. You should use an appropriate mechanism for your environment. DIGEST-MD5 is suitable for test and development environments.
```

Reply:
I wouldn't say only there. In fact you could use PLAIN for dev setups. Maybe GSSAPI would be fine for some environments too. I think the point is that DIGEST-MD5 works just fine but is probably not the most secure.

Reply:
+1 to remove "only"

Reply:
@mposolda Hey, I was thinking about it a little more. SCRAM-SHA-512 is the default SASL mech for Hot Rod. Maybe we should just recommend that and then say other options like GSSAPI are possible but that depends on your environment?

See the `infinispan.client.hotrod.sasl_mechanism` property at: https://docs.jboss.org/infinispan/11.0/apidocs/org/infinispan/client/hotrod/configuration/package-summary.html

Reply:
this is fixed